packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -362,4 +343,300 @@ export class UiControlClient extends ApiCommands {
+   * @property {Relations} params.relation.type - The type of relation.
+   * @property {string} params.relation.text - The text element the relation is based on.
+   */
+  async clickButton(
+    params: {
+      label?: string,
+      relation?: {
+        type: Relations,
+        text: string
+      }
+    },
+  ) {
+    let command = this.click().button();
+
+    if (params.label) {
+      command = command.withText(params.label);
+    }
+
+    if (params.relation) {
+      switch (params.relation.type) {
+        case 'nearestTo':
+          command = command.nearestTo().text(params.relation.text);
+          break;
+        case 'leftOf':
+          command = command.leftOf().text(params.relation.text);
+          break;
+        case 'above':
+          command = command.above().text(params.relation.text);
+          break;
+        case 'rightOf':
+          command = command.rightOf().text(params.relation.text);
+          break;
+        case 'below':
+          command = command.below().text(params.relation.text);
+          break;
+        default:
+          command = command.nearestTo().text(params.relation.text);
+          break;
+      }
+    }
+
+    await command.exec();
+  }
+
+  /**
+   * Click a checkbox with a specific label.
+   * You can also specify where the label is placed relationally.
+   *
+   * **Examples:**
+   * ```typescript
+   * await aui.clickCheckbox({label: 'Toggle'})
+   * await aui.clickCheckbox({label: 'Toggle', relation: {type: 'leftOf'}})
+   * ```
+   *
+   * @param {Object} params - Object containing required `label` property and
+   *                          optional `relation` property.
+   * @property {string} params.label - The label for the checkbox.
+   * @property {Object} [params.relation] - Object describing the relationship between
+   *                                        the clicked checkbox and another element.
+   * @property {Relations} params.relation.type - The type of relation.
+   */
+  async clickCheckbox(
+    params: {
+      label: string,
+      relation?: {
+        type: Relations
+      }
+    },
+  ) {
+    const command = this.click().checkbox();
+
+    if (params.relation) {
+      switch (params.relation.type) {
+        case 'nearestTo':
+          await command.nearestTo().text(params.label).exec();
+          break;
+        case 'leftOf':
+          await command.leftOf().text(params.label).exec();
+          break;
+        case 'above':
+          await command.above().text(params.label).exec();
+          break;
+        case 'rightOf':
+          await command.rightOf().text(params.label).exec();
+          break;
+        case 'below':
+          await command.below().text(params.label).exec();
+          break;
+        default:
+          await command.nearestTo().text(params.label).exec();
+          break;
+      }
+    } else {
+      await command.nearestTo().text(params.label).exec();
+    }
```

Comment:
Minor thing (optional): I think it would help in the future if these convenience functions were implemented more consistently (with `clickButtton()`,  e.g., calling exec always at the end on the command build previously, maybe even sharing some part of the code, e.g., the relationship building part (⟶ currently, switch statement).

Reply:
Very nice approach :)